test/e2e/testdata/data-transformation.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: data-transformation
+  labels:
+    argo-e2e: "true"
+spec:
+  entrypoint: artifact-example
+  templates:
+    - name: artifact-example
+      steps:
+        - - name: generate-artifact
+            template: generate-artifacts
+            withItems: ['script.py', 'doc.pdf', 'foo/script.py', 'foo/doc.pdf']
+            arguments:
+              parameters:
+                - name: file-name
+                  value: "{{item}}"
+        - - name: get-artifact-path
+            template: get-artifact-paths
+        - - name: process-artifact
+            template: process-message
+            withParam: "{{steps.get-artifact-path.outputs.result}}"
+            arguments:
+              artifacts:
+                - name: file
+                  s3:
+                    bucket: test
+                    key: "{{item}}"
+                    endpoint: minio:9000
+                    insecure: true
+                    accessKeySecret:
+                      name: my-minio-cred
+                      key: accesskey
+                    secretKeySecret:
+                      name: my-minio-cred
+                      key: secretkey
+              parameters:
+                - name: file-name
+                  value: "{{item}}"
+
+    - name: generate-artifacts
+      inputs:
+        parameters:
+          - name: file-name
+      container:
+        image: argoproj/argosay:v2
+        args:
+          - echo
+          - hello
+          - "/mnt/{{inputs.parameters.file-name}}"
+      outputs:
+        artifacts:
+          - name: file
+            path: "/mnt/{{inputs.parameters.file-name}}"
+            s3:
```

Comment:
can we add an e2e test that use artifact repository ref please?

Reply:
Sure

Reply:
Done